documentation/add/09_Serializable_Interfaces.md

```diff
@@ -0,0 +1,132 @@
+            end
+        end
+```
+By default, the method `from_old_struct` calls `from_bare_struct` method, so it does not need overloading if the structure of your class public interface have not changed.
+If it have changed, `from_old_struct` needs overloading with the code, converting the old information, stored in the previous class structure into the information, necessary to define new class.
+
+## Interdependent properties problem.
+
+All serializable interface described above and used inside the code is build under assumption that you define class value by assigning the values of the properties one by one. 
+All public setters run checks of the validity of the values one wants to assign to the properties and throw exceptions if the input values are invalid. This assumption causes one problem with interdependent properties. For example, arrays `s`,`e`, and `npix` of a `dnd` or `sqw` object need to be the same shape and size arrays. If you are modifying, say, empty `dnd` object assigning `s`, `e` and `npix` arrays one after another, the first assignment (for `s`) would be always invalid, as the sizes of two other arrays are different from the first one as the initial object was containing empty data in arrays `e` and `npix`.
+
+There is number of ways, one can deal with this issue. After discussion, we have decided to select the following approach, summarized on the **Fig 2** describing the interface  used to validate interdependent properties. 
+
+<center>![serializable Validation Interface](../diagrams/Serializable_validation_interface.png)</center>
+
+**Fig 2:** Validation interface for **serializable** class.
+
+ Any interdependent properties for a child of the *serializable* class, which has interdependent properties, have to overload `[ok,mess,obj] = check_combo_arg(obj)` method. This method verifies validity of interdependent properties. Each setter for interdependent properties runs this method within the following code snippet:
+ ```Matlab
+    function obj = set.an_interdependent_prop(obj,val)
+        check_general_acceptance_of_input_throw_if_invalid(val);
+        if obj.check_combo_arg_
+            [ok,mess,obj] = check_combo_arg(obj,val)
+            if ~ok
+                error('HORACE:interdependent_properties_validation_error',message)
+            end
+        end
+    end
+ ```
+
+Before the properties are set within the serializable class constructor one by one, the *serializable* code sets the protected property **check_combo_arg_** to false, so the check for each interdependent property is not occurring. The code then sets the property **check_combo_arg_** to true and runs the check method: *check_combo_arg* after all interdependent properties have been set, throwing if **check_combo_arg** returns **false**. As the result, user can not change interdependent properties in a wrong way and needs to use constructor or serializable interface to set defined class with interdependent properties.
```

Comment:
I am not clear how check_combo_arg_ gets set true. If all the properties have not been set, how is the use of the class set invalid?

Reply:
its protected property (or private, does not metter), which can be set/unset within the class methods.